### ISSUE TITLE
JGit: Crash on UTF-8 characters in branch name

### ISSUE THREAD

**Issue by realdadfish** [allegro/axion-release-plugin#334]
When my repository contains branch names with UTF-8 characters in it, the release task fails with

```
* What went wrong:
 Execution failed for task ':createRelease'.
 > 
 Malformed input or input contains unmappable characters: /path/torepo/.git/refs/remotes/origin/TF5/dvp-5366-keine-neuen-tickets-vor-klick-auf-abschlie��en
```

I suspect this is a JGit problem, eventually https://bugs.eclipse.org/bugs/show_bug.cgi?id=547400?

**Comment by adamdubiel** [comment]
I bumped JGit to `5.7.x` in `1.12.0` version, maybe the fix is in there?

**Comment by adamdubiel** [comment]
Closing the issue, if bumping version to 5.7 (and in 1.12.1 to 5.8) did not help - i let me know :)

**Comment by realdadfish** [comment]
Sorry, totally forgot about this one. I'll comment if I see this issue again. For now the devs are instructed to avoid UTF-8 and just use plain ASCII :)

**Comment by realdadfish** [comment]
I couldn't find a reference to https://bugs.eclipse.org/bugs/show_bug.cgi?id=547400 in any of the official changelogs (e.g. here https://projects.eclipse.org/projects/technology.jgit/releases/5.7.0), looking at the date at which the fix was checked in it seems to me however that it should have already happened for the timeframe of the 5.5 release.
Since you updated from 5.6 to 5.7 the fix must have either already been included in 5.6 therefor or it was forgotten to be cherry-picked from the "target version 5.1.9" mentioned in the upstream bug.

Anyways, it's kind of hard to reproduce this issue in a live setup without disturbing mechanics, so I'll leave this closed for now and eventually come back to you if it pops up again.
